nixos/tests/librechat.nix

```
@@ -0,0 +1,30 @@
+{
+  name = "librechat";
+
+  nodes.machine = {
+    # Run this test with NIXPKGS_ALLOW_UNFREE=1
```

This doesn't error out without `NIXPKGS_ALLOW_UNFREE=1` and automated tools like r-ryantm can run the tests on a package update.
```suggestion
  node.pkgsReadOnly = false;

  nodes.machine =
    { lib, ... }:
    {
      nixpkgs.config.allowUnfreePredicate =
        pkg:
        builtins.elem (lib.getName pkg) [
          "mongodb"
        ];
        
```

---

It does. It might be inheriting local config if youve allowed either all unfree or "mongodb" specifically.


```
λ gh pr checkout 412668
remote: Enumerating objects: 24, done.
remote: Counting objects: 100% (17/17), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 24 (delta 15), reused 15 (delta 15), pack-reused 7 (from 1)
Unpacking objects: 100% (24/24), 26.00 KiB | 682.00 KiB/s, done.
From github.com:nixos/nixpkgs
 * [new ref]                   refs/pull/412668/head -> librechat-module
Updating files: 100% (24598/24598), done.
Switched to branch 'librechat-module'
λ nix-build -A nixosTests.librechat
error:
       … while evaluating the attribute 'drvPath'
         at /home/jk/projects/personal/nixpkgs/lib/customisation.nix:418:7:
          417|     // {
          418|       drvPath =
             |       ^
          419|         assert condition;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:37:12:
           36|
           37|   strict = derivationStrict drvAttrs;
             |            ^
           38|

       … while evaluating the option `nodes.machine.system.build.toplevel':

       … while evaluating definitions from `/home/jk/projects/personal/nixpkgs/nixos/modules/system/activation/top-level.nix':

       … while evaluating the option `nodes.machine.warnings':

       … while evaluating definitions from `/home/jk/projects/personal/nixpkgs/nixos/modules/system/boot/systemd.nix':

       … while evaluating the option `nodes.machine.systemd.services.mongodb.serviceConfig':

       … while evaluating definitions from `/home/jk/projects/personal/nixpkgs/nixos/modules/services/databases/mongodb.nix':

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: Package ‘mongodb-7.0.16’ in /home/jk/projects/personal/nixpkgs/pkgs/servers/nosql/mongodb/mongodb.nix:186 has an unfree license (‘SSPL’), refusing to evaluate.

       a) To temporarily allow unfree packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNFREE=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnfree = true; }
       in configuration.nix to override this.

       Alternatively you can configure a predicate to allow specific packages:
         { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
             "mongodb"
           ];
         }

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnfree = true; }
       to ~/.config/nixpkgs/config.nix.
```

If you run `nix build .#nixosTests.librechat` it should be more pure and force the error.